lua/src/luabinding.cc

```diff
@@ -278,6 +296,38 @@ int luabinding::write(misc::shared_ptr<io::data> const& data) {
   return retval;
 }
 
+int luabinding::flush() {
+  int retval(0);
+
+  logging::debug(logging::medium) << "lua: luabinding::flush call";
+
+  //For the lua stream connector we only call flush when the queue is full
+  //of unacknowledge events
+  if (has_flush() && multiplexing::muxer::event_queue_max_size() <= _total) {
```

Comment:
IMHO the underlying Lua _flush()_ function should be called whenever the luabinding is called, not only when the event queue max size is reached.

Firstly Centreon Broker calls flush() not only when event queue max size is reached but also when there is no more event to process to improve event freshness. According to me this is a feature.

Secondly I am worried about the duplicated check against the event queue max size that could be bug-prone.